File: GestaoBiblioteca/GestaoBiblioteca/Entities/Emprestimo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace GestaoBiblioteca.Entities;

[Table("tbEmprestimo")]
public class Emprestimo : EntidadePadrao{

    [Column("usuario_id")]
    public int UsuarioId { get; set; }

    [Column("data_emprestimo")]
    public DateTime DataEmprestimo { get; set; }

    [Column("data_devolucao")]
    public DateTime DataDevolucao { get; set; }

    /// <summary>
    /// <para>0 = Em aberto</para>
    /// <para>1 = Devolvido</para>
    /// </summary>
    [Column("status_emprestimo")]
    public bool StatusEmprestimo { get; set; }

    public virtual ICollection<ItensEmprestimo> ItensEmprestimos { get; set; } = new List<ItensEmprestimo>();

    public virtual Usuario Usuario { get; set; } = null!;
}

File: BibliotecaFront/src/app/models/Emprestimo.ts
import { EnumEmprestimoStatus } from './EnumEmprestimoStatus';
import { ItensEmprestimo } from './ItensEmprestimo';
import { Usuario } from './Usuario';
/*

public int UsuarioId { get; set; }

public DateTime DataEmprestimo { get; set; }

public DateTime DataDevolucao { get; set; }

public EnumEmprestimoStatus StatusEmprestimo { get; set; } = EnumEmprestimoStatus.EmAberto;

public virtual ICollection<ItensEmprestimo> ItensEmprestimos { get; set; } = new List<ItensEmprestimo>();

public virtual Usuario Usuario { get; set; } = null!;
*/

export interface Emprestimo {
  id?:string;
  usuarioId?:string;
  usuario?:Usuario;
  dataEmprestimo?:Date;
  dataDevolucao?:Date;
  statusEmprestimo?:EnumEmprestimoStatus;
  itensEmprestimos?:ItensEmprestimo[];
}
